api/envoy/api/v2/route/route_components.proto

```diff
@@ -1174,6 +1174,10 @@ message DirectResponseAction {
   //   :ref:`envoy_api_msg_route.Route`, :ref:`envoy_api_msg_RouteConfiguration` or
   //   :ref:`envoy_api_msg_route.VirtualHost`.
   core.DataSource body = 2;
+
+  // Specifies the response flag to record after sending the response.
+  // If the value is set, it must be a valid %RESPONSE_FLAGS% from access log.
```

Comment:
nit: this is XORed I believe. Might be worth mentioning.

Reply:
See the `ResponseFlags` message in accesslog.proto. Can we use that instead? (It would be nice to move this into a common message but we can do that in a future API version.)

Reply:
@mattklein123 It looks like the `ResponseFlags` is meant for datapath use and not configuration use. 
The externalized names of ResponseFlags differ from what is documented with [access logs output](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log).
```c++
 // Indicates the stream was reset because of a resource overflow.
  bool upstream_overflow = 8;

  // Indicates no route was found for the request.
  bool no_route_found = 9;
```
I can change it to use the list of bools if it's important, but using previously operator externalized names make config easier.

Reply:
Each of this bools corresponds to a response flag code, and this will handle the OR semantics also. Why wouldn't we use it and do the correct translation within the code?

Reply:
These are the names that have been [documented before](https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/accesslog/v3/accesslog.proto#L213). 
`accesslog.ResponseFlagFilter` defines flags as `repeated string` and uses the more well known `{NR, UC, UF,...}` names, can we use those instead ?

Reply:
Yup thanks, `ResponseFlagFilter` sounds great.